VirtoCommerce.CatalogPersonalizationModule.Web/Controllers/Api/PersonalizationModuleController.cs

```diff
@@ -71,9 +106,16 @@ public IHttpActionResult GetTagsCount(string id)
                 EntityId = id,
                 Take = 1
             };
+
             var result = _searchService.SearchTaggedItems(criteria).Results.FirstOrDefault();
             var count = result?.Tags.Count ?? 0;
 
+            var entities = _taggedEntitiesServiceFactory.Create(KnownDocumentTypes.Category).GetEntitiesByIds(new[] { id });
+            if (!entities.IsNullOrEmpty())
+            {
+                var effectiveTagsDict = _tagPropagationPolicy.GetResultingTags(entities);
+                count = effectiveTagsDict[id].Count;
```

Comment:
We need to count unique tags, as own and inherited ones could be duplicated.

Reply:
fixed